src/Microsoft.Diagnostics.EventFlow.Core/Implementations/DiagnosticPipeline.cs

```diff
@@ -97,14 +110,24 @@ public class DiagnosticPipeline : IDisposable
 
             if (sinks.Count > 1)
             {
+                // After broadcasting we will effectively have (sinks.Count - 1) more events in the pipeline.
+                var eventCounter = new TransformBlock<EventData[], EventData[]>(
+                    (batch) => { Interlocked.Add(ref this.eventsInProgress, (sinks.Count - 1) * batch.Length);  return batch; },
```

Comment:
Can you add more comment on this?
This is what I see, and it's even.
NumObservedAtInput - NumDroppedAtFilter - NumOutput = 0

How does this eventCounter work in the calculation of eventInProgress?